### PR TITLE
Add status code to rest.suppressed log output

### PR DESCRIPTION
**Pull request by cbuescher**
Currently the rest.suppressed logger in RestResponse logs the request path and the request parameters for 500 errors (warn level) and 400s (debug). In order to be able to filter on those status codes more efficiently we should add them to the log message.
This change also adds testing around the logic deciding when we log with respect to the request parameters.

### CODE REVIEW

**Comment by cbuescher** [comment]
Just as a note, as a follow up we'd like to remove the condition to not log when the "error_trace" parameter is set by the client, in order to have a full record of errors in the logs for monitoring.

File: server/src/main/java/org/elasticsearch/rest/RestResponse.java [hunk 1]
```diff
@@ -120,9 +120,10 @@ public RestResponse(RestChannel channel, RestStatus status, Exception e) throws
             // log exception only if it is not returned in the response
             Supplier<?> messageSupplier = () -> String.format(
                 Locale.ROOT,
-                "path: %s, params: %s",
+                "path: %s, params: %s, status: %s",
```
Comment (quux00) [inline server/src/main/java/org/elasticsearch/rest/RestResponse.java:123]: Minor nit: should we use `%d` formatter instead of `%s` since status is an int?

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Review by pgomulka — APPROVED** [review]
LGTM

**Review by quux00 — APPROVED** [review]
LGTM

**Pull request merged**